From 635b96a7944697bddf0254b9158a76dd1afe9840 Mon Sep 17 00:00:00 2001
From: Varun Vasudev <vvasudev@apache.org>
Date: Fri, 11 Sep 2015 14:37:10 +0530
Subject: [PATCH 1135/2848] YARN-4136. LinuxContainerExecutor loses info when
 forwarding ResourceHandlerException. Contributed
 by Bibin A Chundatt.

(cherry picked from commit 486d5cb803efec7b4db445ee65a3df83392940a3)
(cherry picked from commit a4288f1262832bb997bab4c98f072366a84a686b)

Change-Id: I74ff9010a68036f30d4dd0e127dfeb4bd28fae1a
---
 .../server/nodemanager/LinuxContainerExecutor.java |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java
index 6a348f1..ba1a85f 100644
--- a/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java
+++ b/hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-nodemanager/src/main/java/org/apache/hadoop/yarn/server/nodemanager/LinuxContainerExecutor.java
@@ -331,7 +331,7 @@ public int launchContainer(Container container,
       }
     } catch (ResourceHandlerException e) {
       LOG.error("ResourceHandlerChain.preStart() failed!", e);
-      throw new IOException("ResourceHandlerChain.preStart() failed!");
+      throw new IOException("ResourceHandlerChain.preStart() failed!", e);
     }
 
     ShellCommandExecutor shExec = null;
-- 
1.7.9.5

